py/lexer.c

```diff
@@ -119,6 +125,37 @@ STATIC bool is_tail_of_identifier(mp_lexer_t *lex) {
     return is_head_of_identifier(lex) || is_digit(lex);
 }
 
+STATIC void swap_char_banks(mp_lexer_t *lex) {
+    if (lex->vstr_postfix_processing) {
+        unichar h0, h1, h2;
+
+        h0 = lex->chr0;
+        h1 = lex->chr1;
+        h2 = lex->chr2;
+
+        lex->chr0 = lex->vstr_postfix.len > 0 ? lex->vstr_postfix.buf[0] : 0;
+        lex->chr1 = lex->vstr_postfix.len > 1 ? lex->vstr_postfix.buf[1] : 0;
+        lex->chr2 = lex->vstr_postfix.len > 2 ? lex->vstr_postfix.buf[2] : 0;
+        lex->chr3 = h0;
+        lex->chr4 = h1;
+        lex->chr5 = h2;
+
+        lex->vstr_postfix_idx = lex->vstr_postfix.len > 2 ? 3 : lex->vstr_postfix.len;
+    } else {
+        // blindly reset to the "backup" bank when done postfix processing
+        // this restores control to the mp_reader
+        lex->chr0 = lex->chr3;
+        lex->chr1 = lex->chr4;
+        lex->chr2 = lex->chr5;
+        lex->chr3 = 0;
+        lex->chr4 = 0;
+        lex->chr5 = 0;
+
+        vstr_reset(&lex->vstr_postfix);
+        lex->vstr_postfix_idx = 0;
```

Comment:
Does idx need to be reset?  Do the vstr need to be reset?

Reply:
Given the current implementation: I think since I blindly write `.format(` and the postfix translations to `vstr_postfix` at least the string should be reset/cleared. `vstr_postfix_idx` needs reset *somewhere*, though there may well be better places to do it.

If you're implying it'd be lighter on code size/CPU cycles/memory usage to just maintain a constantly growing `vstr` that starts to look roughly like `.format(x).format(y).format(z)`, with an ever-increasing `vstr_postfix_idx`, I'm totally open to trying that. I'm not sure how expensive the `vstr_reset`s are

Reply:
The reset is cheap, I just thought it might be better to reset it before use (to be guaranteed it's reset), rather than after use (when you may not use it again).